rfc/cdevents-spinnaker.md

```diff
@@ -0,0 +1,149 @@
+# Implementing CDEvents to Spinnaker using Java SDK
+
+
+| | |
+|-|-|
+| **Status**     | _**Proposed**, Accepted, Implemented, Obsolete_ |
+| **RFC #**      | _#299_ |
+| **Author(s)**  | _Jalander Ramagiri (`@rjalander`)_|
+| **SIG / WG**   | CDF-CDEvents, CDF-SIGEvents |
+
+## Overview
+
+Many organisations use more than one CI/CD tools as part of their pipelines and most of the times the organisations implement their own glue code in order to make the tools work with each other.
+
+Implementing [CDEvents](https://cdevents.dev/) to Spinnaker enables Spinnaker to communicate with other CI/CD tools with in the ecosystem. [CDEvents](https://cdevents.dev/) is a common specification for Continuous Delivery events based on CloudEvents.
+Spinnaker can produce and consume [CDEvents](https://cdevents.dev/) using [Java SDK](https://github.com/cdevents/sdk-java) which is developed and maintained by the [CDEvents community](https://cdevents.dev/community/).
+
+## Goals and Non-Goals
+Goals:
+- Event-Driven Standardization with CI/CD tools like Keptn, Tekton, Jenkins, Spinnaker, etc..
+- Spinnaker can able to consume and produce CDEvents from/to `events-broker`
```

Comment:
Spinnaker as it sits has no concept of an event bus. This would require reworking the current architecture on how Spinnaker could handle these events. There's this notion of a context that houses all events that is collected at the Orca service. This means Orca would need to somehow organize and re-build the context as the each stage/task/operation context is usually propagated back to Orca and is built that way, but with the event bus, that seems a little bit more tricky.

Reply:
Thanks @xibz, for sharing your comments. 
As per this design goals this will be achieved by using an existing Webhook and Pub/Sub Automated triggers, Spinnaker can able to consume and produce CDEvents from/to `events-broker`

Reply:
Thanks @rjalander. I think what might help is a diagram with multiple systems, eg some CI service, some CD service, and illustrate where the event bus(es) sit(s). Once we have that diagram it'll remove our assumptions, and allow us to directly ask questions against it.